### PR TITLE
chore: Update datafusion 

### PR DESCRIPTION
**Pull request by alamb**
# Rationale:
Keep up with DataFusion. Notably https://github.com/apache/arrow-datafusion/pull/1455 and https://github.com/apache/arrow-datafusion/issues/1476

# Changes:
1. Update DataFusion pin to https://github.com/apache/arrow-datafusion/commit/401271377cd84dc1546827f66bda1b242860a6a8

### CODE REVIEW

File: datafusion_util/src/lib.rs [hunk 1]
```diff
@@ -52,8 +52,8 @@ impl AsExpr for Expr {
 pub fn make_range_expr(start: i64, end: i64, time: impl AsRef<str>) -> Expr {
     // We need to cast the start and end values to timestamps
     // the equivalent of:
-    let ts_start = ScalarValue::TimestampNanosecond(Some(start));
-    let ts_end = ScalarValue::TimestampNanosecond(Some(end));
+    let ts_start = ScalarValue::TimestampNanosecond(Some(start), None);
```
Comment (alamb) [inline datafusion_util/src/lib.rs:55]: this is due to the `ScalarValues` in DataFusion gaining a timezone

File: query/src/provider.rs [hunk 2]
```diff
@@ -2483,7 +2484,14 @@ mod test {
 
             let schema: Schema = batch.schema().try_into().unwrap();
             for field_idx in 0..schema.len() {
-                assert!(schema.field(field_idx).0.is_some());
+                let (influx_column_type, field) = schema.field(field_idx);
```
Comment (alamb) [inline query/src/provider.rs:2487]: this test was failing (due to https://github.com/apache/arrow-datafusion/issues/1476) so I improved the diagnostic output when it failed

File: query_tests/cases/in/delete_multi_expr_one_chunk.expected [hunk 3]
```diff
@@ -1,19 +1,19 @@
 -- Test Setup: OneDeleteMultiExprsOneChunk
--- SQL: SELECT * from cpu;
+-- SQL: SELECT * from cpu order by bar, foo, time;
```
Comment (alamb) [inline query_tests/cases/in/delete_multi_expr_one_chunk.expected:2]: Some of these queries had 'jittery' output (aka the output changed from run to run) as the output order was not specified. 

So I changed the queries to make the output order deterministic. Also, found https://github.com/apache/arrow-datafusion/issues/1479 while I was at it 😆

File: query_tests/cases/in/pushdown.expected [hunk 4]
```diff
@@ -165,16 +165,16 @@
...
+|               |     FilterExec: 5 < system@1 AND CAST(town@3 AS Utf8) != tewsbury AND system@1 < 7 AND CAST(count@0 AS Int64) = 632 OR CAST(town@3 AS Utf8) = reading AND time@2 > 130                                                                                                                             |
+|               |       RepartitionExec: partitioning=RoundRobinBatch(4)                                                                                                                                                                                                                                             |
+|               |         IOxReadFilterNode: table_name=restaurant, chunks=0 predicate=Predicate exprs: [Float64(5) < #system, #town != Utf8("tewsbury"), #system < Float64(7), #time > TimestampNanosecond(130, None)]                                                                                              |
```
Comment (alamb) [inline query_tests/cases/in/pushdown.expected:178]: the difference is the `None` timezone has appeared in the explain plan output

File: workspace-hack/Cargo.toml [hunk 5]
```diff
@@ -49,7 +48,6 @@ tracing-core = { version = "0.1", features = ["lazy_static", "std"] }
 tracing-subscriber = { version = "0.3", features = ["alloc", "ansi", "ansi_term", "env-filter", "fmt", "lazy_static", "matchers", "regex", "registry", "sharded-slab", "smallvec", "std", "thread_local", "tracing", "tracing-log"] }
 
 [build-dependencies]
-ahash = { version = "0.7", features = ["std"] }
```
Comment (carols10cents) [inline workspace-hack/Cargo.toml:52]: 🎉

**Review by carols10cents — APPROVED** [review]
LGTM!

**Review by pauldix — APPROVED** [review]

**Pull request merged**